libkbfs/keybase_service_base.go

```diff
@@ -440,16 +448,58 @@ func (k *KeybaseServiceBase) LoadUserPlusKeys(ctx context.Context,
 // LoadTeamPlusKeys implements the KeybaseService interface for
 // KeybaseServiceBase.
 func (k *KeybaseServiceBase) LoadTeamPlusKeys(
-	ctx context.Context, tid keybase1.TeamID) (TeamInfo, error) {
+	ctx context.Context, tid keybase1.TeamID, desiredKeyGen KeyGen,
+	desiredUser keybase1.UserVersion, desiredRole keybase1.TeamRole) (
+	TeamInfo, error) {
+	if desiredRole != keybase1.TeamRole_NONE &&
+		desiredRole != keybase1.TeamRole_WRITER &&
+		desiredRole != keybase1.TeamRole_READER {
+		panic("KBFS shouldn't care about non-writer/reader roles")
```

Comment:
But we're allowing `TeamRole_NONE`? Also, can we create a static `map[keybase1.TeamRole]bool` for allowed team roles that we can check rather than doing a 3-way comparison?